core/src/main/java/org/apache/brooklyn/core/objs/AbstractBrooklynObject.java

```diff
@@ -55,8 +57,18 @@
     @SetFromFlag(value = "tags")
     private final Set<Object> tags = Sets.newLinkedHashSet();
 
+    private RelationSupportInternal<PublicSelfType> relations = new ByObjectBasicRelationSupport<PublicSelfType>(getPublicThis(), new RelationChangedCallback());
```

Comment:
the no-arg constructor *is* invoked during rebind.  i've added a test to confirm (and assert the sequence).